transport/transport.go

```diff
@@ -689,6 +672,12 @@ var (
 	// ErrStreamDrain indicates that the stream is rejected by the server because
 	// the server stops accepting new RPCs.
 	ErrStreamDrain = streamErrorf(codes.Unavailable, "the server stops accepting new RPCs")
```

Comment:
Do we need this anymore?

IIRC, the purpose of this error is exactly what `Unprocessed()` is for.

And if we no longer need this error, we can also (maybe) remove `s.goaway` channel.
But should probably make this a TODO for another cleanup PR.

Reply:
Unexported, added TODO.  We have a lot more cleaning up we can do.